avatar/urls.py

```diff
@@ -1,18 +1,23 @@
-from django.urls import re_path
+from django.urls import path, re_path
 
 from avatar import views
 
+# For reverseing namespaced urls
+# https://docs.djangoproject.com/en/4.1/topics/http/urls/#reversing-namespaced-urls
+app_name = "avatar"
+
 urlpatterns = [
-    re_path(r"^add/$", views.add, name="avatar_add"),
-    re_path(r"^change/$", views.change, name="avatar_change"),
-    re_path(r"^delete/$", views.delete, name="avatar_delete"),
-    re_path(
-        r"^render_primary/(?P<user>[\w\d\@\.\-_]+)/(?P<width>[\d]+)/$",
+    path("add/", views.add, name="avatar_add"),
+    path("change/", views.change, name="avatar_change"),
+    path("delete/", views.delete, name="avatar_delete"),
+    # https://docs.djangoproject.com/en/4.1/topics/http/urls/#path-converters
+    path(
+        "render_primary/<slug:user>/<int:width>/",
         views.render_primary,
         name="avatar_render_primary",
     ),
     re_path(
-        r"^render_primary/(?P<user>[\w\d\@\.\-_]+)/(?P<width>[\d]+)/(?P<height>[\d]+)/$",
+        "render_primary/<slug:user>/<int:width>/<int:height>",
```

Comment:
@0xMRTT Is there a particular reason why we keep `re_path` here? And it looks like we remove the `/` at the end?

Reply:
Oh my bad, missed to remove it 🙃